File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp" 

using namespace std::placeholders;  // for "_1"
     
class SmartphoneNode : public rclcpp::Node { 
    public:
        SmartphoneNode() : Node("smartphone") { // MODIFY NAME
            subscriber_ =this->create_subscription<example_interfaces::msg::String>("robot_news", 10, std::bind(&SmartphoneNode::callBackRobotNews, this, _1)); // create a subscription to the "robot_news" topic
            RCLCPP_INFO(this->get_logger(), "Smartphone node has been started from CPP!");
        }
     
    private:
        void callBackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
            RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str()); // log the received message
        }
        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_; // Subscription object
};
     
int main(int argc, char **argv){
    rclcpp::init(argc, argv); // initialize ROS2
    auto node = std::make_shared<SmartphoneNode>(); // Create Node
    rclcpp::spin(node); // keep the node alive to listen for messages
    rclcpp::shutdown(); // Shutdown ROS2
    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
     
     
class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")  #  Name of the node
        self.subscriber = self.create_subscription(String, "robot_news", self.callback_robot_news, 10) # create a subscription to the "robot_news" topic
        self.get_logger().info("Smartphone node has been started from python!") 
        
    def callback_robot_news(self, msg: String):
        self.get_logger().info(msg.data) # log the received message

     
def main(args=None):
    rclpy.init(args=args) # initialize rclpy
    node = SmartphoneNode() # create node instance
    rclpy.spin(node) # keep node alive to listen to messages
    rclpy.shutdown()
     
if __name__ == "__main__":
    main()
